### PR TITLE
OpenCensus Cloud Spanner integration samples

### PR DESCRIPTION
**Pull request by mayurkale22**
- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [x] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/opencensus/src/java/com/example/spanner/opencensus/GrpcMetricSample.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 2) {
+      System.err.println("Usage: GrpcMetricSample <instance_id> <database_id>");
+      return;
+    }
+
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    String instanceId = args[0];
```
Comment (shubha-rajan) [inline spanner/opencensus/src/java/com/example/spanner/opencensus/GrpcMetricSample.java:44]: Command line arguments are an anti-pattern in samples. The sample classes should include a public method that contains the logic of the sample, which is called by the main method. This makes it easier to test samples and run them from an IDE. See the example in [SAMPLE_FORMAT.md](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md) for more details on what sample structure should look like.

File: spanner/opencensus/src/java/com/example/spanner/opencensus/QueryStatsMetricSample.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 2) {
+      System.err.println("Usage: QueryStatsMetricSample <instance_id> <database_id>");
+      return;
+    }
+
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    String instanceId = args[0];
```
Comment (shubha-rajan) [inline spanner/opencensus/src/java/com/example/spanner/opencensus/QueryStatsMetricSample.java:85]: See comment in other file about sample format

**Review by shubha-rajan — CHANGES_REQUESTED** [review]
Thanks for opening this PR. Please take another look at [SAMPLE_FORMAT.md](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md) and update this sample to meet those standards. Also, before this sample can be merged, you will need to add integration tests for both of the new classes. Please let us know if you need any help setting those up.

File: spanner/opencensus/src/java/com/example/spanner/opencensus/GrpcMetricSample.java (hunk 1 above)
Comment (mayurkale22) [reply spanner/opencensus/src/java/com/example/spanner/opencensus/GrpcMetricSample.java]: Done, ptal

File: spanner/opencensus/src/java/com/example/spanner/opencensus/QueryStatsMetricSample.java (hunk 2 above)
Comment (mayurkale22) [reply spanner/opencensus/src/java/com/example/spanner/opencensus/QueryStatsMetricSample.java]: done, ptal

**Comment by mayurkale22** [comment]
@shubha-rajan I see some trouble with the test, could you help solve it?

**Comment by shubha-rajan** [comment]
> @shubha-rajan I see some trouble with the test, could you help solve it?

Yeah! I'll take a look right now.

**Comment by shubha-rajan** [comment]
@mayurkale22 this is the error message:
```
[ERROR] COMPILATION ERROR :
[ERROR] /tmpfs/src/github/java-docs-samples/spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java:[120,5] cannot find symbol
  symbol:   variable GrpcMetricSample
  location: class com.example.spanner.opencensus.GrpcMetricSampleIT
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project cloud-spanner-opencensus: Compilation failure
[ERROR] /tmpfs/src/github/java-docs-samples/spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java:[120,5] cannot find symbol
[ERROR]   symbol:   variable GrpcMetricSample
[ERROR]   location: class com.example.spanner.opencensus.GrpcMetricSampleIT
[ERROR] -> [Help 1]
```
The problem is your directory structure. You'll want to update it to look like this: 
```
├── README.md
├── pom.xml
└── src
    └── java
        ├── main
        │   └── com
        │       └── example
        │           └── spanner
        │               └── opencensus
        │                   ├── GrpcMetricSample.java
        │                   └── QueryStatsMetricSample.java
        └── test
            └── java
                └── com
                    └── example
                        └── spanner
                            └── opencensus
                                └── GrpcMetricSampleIT.java
```

**Comment by mayurkale22** [comment]
@shubha-rajan thanks for the help! The tests are in place, please review the PR as time permits

File: spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    final String out = runExample(client);
+
+    assertThat(out).contains("1 1 Title 1");
+  }
+
+  private String runExample(DatabaseClient client) {
+    PrintStream stdOut = System.out;
+    ByteArrayOutputStream bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
```
Comment (shubha-rajan) [inline spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java:120]: Can you put this in a separate method with the @Before annotation?

File: spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    GrpcMetricSample.captureGrpcMetric(client);
+    System.setOut(stdOut);
+    return bout.toString();
```
Comment (shubha-rajan) [inline spanner/opencensus/src/test/java/com/example/spanner/opencensus/GrpcMetricSampleIT.java:123]: Same as above, move this to a separate method with the @After annotation

File: spanner/opencensus/src/test/java/com/example/spanner/opencensus/QueryStatsMetricSampleIT.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  private String runExample(DatabaseClient client) {
+    PrintStream stdOut = System.out;
+    ByteArrayOutputStream bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
+    QueryStatsMetricSample.captureQueryStatsMetric(client);
+    System.setOut(stdOut);
+    return bout.toString();
```
Comment (shubha-rajan) [inline spanner/opencensus/src/test/java/com/example/spanner/opencensus/QueryStatsMetricSampleIT.java:122]: Same comment as in the other file, please move the setup and teardown steps to separate methods with the @Before and @After annotations. Example:

```
  @Before
  public void captureOutput() {
    bout = new ByteArrayOutputStream();
    System.setOut(new PrintStream(bout));
  }

  @After
  public void resetOutput() {
    System.setOut(originalOut);
    bout.reset();
  }
```

**Review by shubha-rajan — CHANGES_REQUESTED** [review]

**Comment by shubha-rajan** [comment]
HI @mayurkale22 I just took another pass at reviewing. PTAL and tag me for another review when you're ready!

File: spanner/opencensus/src/main/java/com/example/spanner/opencensus/GrpcMetricSample.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Statement;
+import io.opencensus.contrib.grpc.metrics.RpcViews;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
+
+/**
+ * This sample demonstrates how to record and export client round-trip latency using OpenCensus.
+ */
+public class GrpcMetricSample {
```
[the remark this answers is not in the document]
Comment (mayurkale22) [reply spanner/opencensus/src/main/java/com/example/spanner/opencensus/GrpcMetricSample.java]: Done

File: spanner/opencensus/src/main/java/com/example/spanner/opencensus/GrpcMetricSample.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
```
[the remark this answers is not in the document]
Comment (mayurkale22) [reply spanner/opencensus/src/main/java/com/example/spanner/opencensus/GrpcMetricSample.java]: Done

File: spanner/opencensus/src/test/java/com/example/spanner/opencensus/QueryStatsMetricSampleIT.java (hunk 5 above)
Comment (mayurkale22) [reply spanner/opencensus/src/test/java/com/example/spanner/opencensus/QueryStatsMetricSampleIT.java]: Done

**Comment by mayurkale22** [comment]
> HI @mayurkale22 I just took another pass at reviewing. PTAL and tag me for another review when you're ready!

Thanks for reviews and suggestions. I made the changes, PTAL

**Review by shubha-rajan — APPROVED** [review]
LGTM! Thank you for making those changes!

**Pull request merged**